api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -36,6 +39,15 @@ const (
 	defaultAKSNodeSubnetCIDRForOverlay = "10.224.0.0/16"
 )
 
+// setDefaultResourceGroupName sets the default ResourceGroupName for an AzureManagedControlPlane.
+func (m *AzureManagedControlPlane) setDefaultResourceGroupName() {
```

Comment:
is this new defaulting?

Reply:
Yes. I lost track of the exact part in the code this was needed, but the controlle expects there to be a resource group name set and caused the E2E test to fail otherwise. Will try and find the exact issue and update it here